test/e2e/main_test.go

```diff
@@ -412,6 +412,59 @@ func TestOperatorUpgrade(t *testing.T) {
 	}
 }
 
+// TestIsManagedByController test prometheus operator managing object with correct ControlerID.
+func TestIsManagedByController(t *testing.T) {
+	skipPrometheusTests(t)
+
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+
+	ns := framework.CreateNamespace(context.Background(), t, testCtx)
+
+	// Create operator-1.
+	finalizers, err := framework.CreateOrUpdatePrometheusOperator(context.Background(), ns, nil, nil, nil, nil, true, true, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, f := range finalizers {
+		testCtx.AddFinalizerFn(f)
+	}
+
+	ns = framework.CreateNamespace(context.Background(), t, testCtx)
+	// Create operator-2 in a new ns and set controller-id.
+	addArgs := []string{"--controller-id=42"}
+	finalizers, err = framework.CreateOrUpdatePrometheusOperatorWithOpts(context.Background(),
+		operatorFramework.PrometheusOperatorOpts{
+			Namespace:              ns,
+			AllowedNamespaces:      nil,
+			DeniedNamespaces:       nil,
+			PrometheusNamespaces:   nil,
+			AlertmanagerNamespaces: nil,
+			EnableAdmissionWebhook: false,
+			ClusterRoleBindings:    true,
+			EnableScrapeConfigs:    true,
+			AdditionalArgs:         addArgs,
+		})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, f := range finalizers {
+		testCtx.AddFinalizerFn(f)
+	}
+
+	testFuncs := map[string]func(t *testing.T){
+		"MultipleOperatorsPrometheusServer": testMultipleOperatorsPrometheusServer,
+		"MultipleOperatorsPrometheusAgent":  testMultipleOperatorsPrometheusAgent,
+		"MultipleOperatorsAlertManager":     testMultipleOperatorsAlertManager,
+		"MultipleOperatorsThanosRuler":      testMultipleOperatorsThanosRuler,
```

Comment:
(nit) we can shorten a bit
```suggestion
		"PrometheusServer": testMultipleOperatorsPrometheusServer,
		"PrometheusAgent":  testMultipleOperatorsPrometheusAgent,
		"Alertmanager":     testMultipleOperatorsAlertManager,
		"ThanosRuler":      testMultipleOperatorsThanosRuler,
```

Reply:
done